File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/git_folder_metadatum.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class GitFolderMetadatum(object):
    """
    Details about a git repository
    """

    def __init__(self, **kwargs):
        """
        Initializes a new GitFolderMetadatum object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param repo_key:
            The value to assign to the repo_key property of this GitFolderMetadatum.
        :type repo_key: str

        :param is_associated:
            The value to assign to the is_associated property of this GitFolderMetadatum.
        :type is_associated: bool

        """
        self.swagger_types = {
            'repo_key': 'str',
            'is_associated': 'bool'
        }

        self.attribute_map = {
            'repo_key': 'repoKey',
            'is_associated': 'isAssociated'
        }

        self._repo_key = None
        self._is_associated = None

    @property
    def repo_key(self):
        """
        **[Required]** Gets the repo_key of this GitFolderMetadatum.
        Unique repo key if folder/file path is associated with a git folder


        :return: The repo_key of this GitFolderMetadatum.
        :rtype: str
        """
        return self._repo_key

    @repo_key.setter
    def repo_key(self, repo_key):
        """
        Sets the repo_key of this GitFolderMetadatum.
        Unique repo key if folder/file path is associated with a git folder


        :param repo_key: The repo_key of this GitFolderMetadatum.
        :type: str
        """
        self._repo_key = repo_key

    @property
    def is_associated(self):
        """
        **[Required]** Gets the is_associated of this GitFolderMetadatum.
        If the passed folder/file path is associated with a git folder. (Active/Inactive)


        :return: The is_associated of this GitFolderMetadatum.
        :rtype: bool
        """
        return self._is_associated

    @is_associated.setter
    def is_associated(self, is_associated):
        """
        Sets the is_associated of this GitFolderMetadatum.
        If the passed folder/file path is associated with a git folder. (Active/Inactive)


        :param is_associated: The is_associated of this GitFolderMetadatum.
        :type: bool
        """
        self._is_associated = is_associated

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/GitFolderMetadatum.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Details about a Git repository.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=GitFolderMetadatum.Builder.class)

public final class GitFolderMetadatum  {
    @Deprecated
    @java.beans.ConstructorProperties({"repoKey", "isAssociated"})
    public GitFolderMetadatum(String repoKey, Boolean isAssociated) {
        super();
        this.repoKey = repoKey;
        this.isAssociated = isAssociated;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Unique repo key if folder/file path is associated with a Git folder.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("repoKey")
        private String repoKey;

                /**
         * Unique repo key if folder/file path is associated with a Git folder.
         * @param repoKey the value to set
         * @return this builder
         **/
        

        public Builder repoKey(String repoKey) {
        this.repoKey = repoKey;
        return this;
        }
            /**
     * If the passed folder/file path is associated with a Git folder. (Active/Inactive)
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("isAssociated")
        private Boolean isAssociated;

                /**
         * If the passed folder/file path is associated with a Git folder. (Active/Inactive)
         * @param isAssociated the value to set
         * @return this builder
         **/
        

        public Builder isAssociated(Boolean isAssociated) {
        this.isAssociated = isAssociated;
        return this;
        }


        public GitFolderMetadatum build() {
            GitFolderMetadatum model = new GitFolderMetadatum(this.repoKey
                    , this.isAssociated);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(GitFolderMetadatum model) {
                this.repoKey(model.getRepoKey());
    this.isAssociated(model.getIsAssociated());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Unique repo key if folder/file path is associated with a Git folder.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("repoKey")
    private final String repoKey;

        /**
     * Unique repo key if folder/file path is associated with a Git folder.
     * @return the value
     **/
    
    public String getRepoKey() {
        return repoKey;
    }


        /**
     * If the passed folder/file path is associated with a Git folder. (Active/Inactive)
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isAssociated")
    private final Boolean isAssociated;

        /**
     * If the passed folder/file path is associated with a Git folder. (Active/Inactive)
     * @return the value
     **/
    
    public Boolean getIsAssociated() {
        return isAssociated;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("GitFolderMetadatum(");
        sb.append("repoKey=").append(String.valueOf(this.repoKey));
        sb.append(", isAssociated=").append(String.valueOf(this.isAssociated));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof GitFolderMetadatum)) {
            return false;
        }

        GitFolderMetadatum other = (GitFolderMetadatum) o;
        return java.util.Objects.equals(this.repoKey, other.repoKey) &&
            java.util.Objects.equals(this.isAssociated, other.isAssociated);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.repoKey == null ? 43 : this.repoKey.hashCode());
        result = (result * PRIME) + (this.isAssociated == null ? 43 : this.isAssociated.hashCode());
        return result;
    }


}
